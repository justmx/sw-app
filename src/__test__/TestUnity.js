import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";

export const ReduxProvider = ({ children, initialState = {} }) => {
  const middlewares = [thunk];
  const mockStore = configureMockStore(middlewares);
  const store = mockStore(initialState);
  return <Provider store={store}>{children}</Provider>;
};
