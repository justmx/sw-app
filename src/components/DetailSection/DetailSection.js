import React from "react";

const DetailSection = ({ person }) => {
  if (!person) return null;
  const { name, height, birth_year, gender, allfilms } = person;

  return (
    <div className="swap-detail-container">
      <h2>Detail section</h2>
      <div className="swap-detail">
        <div className="swap-detail-info">
          Name: <span>{name}</span>
        </div>
        <div className="swap-detail-info">
          Height: <span>{height}</span>
        </div>
        <div className="swap-detail-info">
          Birth year: <span>{birth_year}</span>
        </div>
        <div className="swap-detail-info">
          Gender: <span>{gender}</span>
        </div>
        <div className="swap-detail-info">
          List of films:
          {allfilms.map((film, i) => {
            return <li key={i}>{film.title}</li>;
          })}
        </div>
      </div>
    </div>
  );
};

export default DetailSection;
