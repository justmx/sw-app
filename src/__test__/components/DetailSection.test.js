import React from "react";
import { shallow, mount } from "enzyme";
import { ReduxProvider } from "../TestUnity";
import DetailSection from "../../components/DetailSection/DetailSection";

const person = {
  name: "Jack",
  height: "177",
  mass: 168,
  allfilms: [
    { title: "The Empire Strikes Back" },
    { title: "Return of the Jedi" },
  ],
};

describe("Render DetailSection", () => {
  it("should render DetailSection", () => {
    const wrapper = shallow(<DetailSection person={person} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should return null if no props", () => {
    const wrapper = shallow(<DetailSection />);
    expect(wrapper.type()).toEqual(null);
  });
});

describe("Render person details", () => {
  const wrapper = mount(<DetailSection person={person} />);

  it("check prop name", () => {
    expect(wrapper.props().person).toHaveProperty("name");
  });

  it("should render person name", () => {
    expect(wrapper.find(".swap-detail-info").at(0).text()).toEqual(
      "Name: Jack"
    );
  });

  it("check person films", () => {
    expect(wrapper.find("li")).toHaveLength(2);
    expect(wrapper.find("li").at(0).text()).toEqual("The Empire Strikes Back");
  });
});
