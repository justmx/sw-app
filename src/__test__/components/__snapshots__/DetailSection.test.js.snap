// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render DetailSection should render DetailSection 1`] = `
<div
  className="swap-detail-container"
>
  <h2>
    Detail section
  </h2>
  <div
    className="swap-detail"
  >
    <div
      className="swap-detail-info"
    >
      Name: 
      <span>
        Jack
      </span>
    </div>
    <div
      className="swap-detail-info"
    >
      Height: 
      <span>
        177
      </span>
    </div>
    <div
      className="swap-detail-info"
    >
      Birth year: 
      <span />
    </div>
    <div
      className="swap-detail-info"
    >
      Gender: 
      <span />
    </div>
    <div
      className="swap-detail-info"
    >
      List of films:
      <li
        key="0"
      >
        The Empire Strikes Back
      </li>
      <li
        key="1"
      >
        Return of the Jedi
      </li>
    </div>
  </div>
</div>
`;
